arthur/tasks.py

```diff
@@ -228,26 +235,67 @@ def get(self, task_id):
         :raises NotFoundError: raised when the requested task is not
             found on the registry
         """
-        try:
-            self._rwlock.reader_acquire()
-            task = self._tasks[task_id]
-        except KeyError:
+        self._rwlock.reader_acquire()
+        found = self.conn.exists(task_id)
+        self._rwlock.reader_release()
+
+        if not found:
             raise NotFoundError(element=str(task_id))
-        finally:
-            self._rwlock.reader_release()
+
+        self._rwlock.reader_acquire()
+        task_dump = self.conn.get(task_id)
+        task = pickle.loads(task_dump)
+        self._rwlock.reader_release()
 
         return task
 
+    def update(self, task_id, task):
+        """Update a task in the registry.
+
+        Update a task stored in the registry using its task identifier. When
+        the task does not exist, a `NotFoundError` exception will be
+        raised.
+
+        :param task_id: task identifier
+        :param task: task object
+
+        :returns: a task object
+        """
+        self._rwlock.reader_acquire()
+        found = self.conn.exists(task_id)
+        self._rwlock.reader_release()
+
+        if not found:
+            logger.warning("Task %s not found, adding it", str(task_id))
+
+        self._rwlock.writer_acquire()
+        self.conn.set(task_id, pickle.dumps(task))
+        self._rwlock.writer_release()
+
+        logger.debug("Task %s updated", str(task_id))
+
     @property
     def tasks(self):
         """Get the list of tasks"""
 
         self._rwlock.reader_acquire()
-        tl = [v for v in self._tasks.values()]
-        tl.sort(key=lambda x: x.task_id)
+        tasks = []
+        keys = []
+
+        total, found = self.conn.scan()
+        keys.extend([f.decode("utf-8") for f in found])
+        while total != 0:
+            total, found = self.conn.scan(cursor=total)
```

Comment:
I realized this will return all the objects stored in the database. In this case, jobs, messages, etc. I think tasks should have an internal prefix. RQ uses a prefix, so probably we can use the same. For example, when you try to access a job with a hash, you use that hash but internally RQ looks for keys with the value `rq:Job:<hash>` or something similar.

We will need a test for this case too, where there's something stored in the database that is not a task and this method returns only the tasks.